python/grass/temporal/abstract_space_time_dataset.py

```diff
@@ -1834,43 +1848,61 @@ def get_registered_maps(
             else:
                 map_view = self.get_new_map_instance(None).get_type() + "_view_rel_time"
 
-            if columns is not None and columns != "":
-                sql = "SELECT %s FROM %s  WHERE %s.id IN (SELECT id FROM %s)" % (
-                    columns,
-                    map_view,
-                    map_view,
-                    self.get_map_register(),
-                )
+            if group:
+                if not columns:
+                    columns = f"{group},group_concat(id,',') AS id"
+                else:
+                    columns = (
+                        group
+                        + ", "
+                        + ", ".join(
+                            [
+                                f"group_concat({column},',') AS {column}"
+                                for column in columns.split(",")
+                            ]
+                        )
+                    )
+                if order is not None and order != "":
+                    order = f"{group},{order.split(';')[0]}"
+
             else:
-                sql = "SELECT * FROM %s  WHERE %s.id IN (SELECT id FROM %s)" % (
-                    map_view,
-                    map_view,
-                    self.get_map_register(),
-                )
+                columns = columns or "*"
```

Comment:
What does this line do? does it return a boolean? It can be the first usage of the function's "column" argument if group takes the else path.

Reply:
Thanks for looking into this.

This line assures that all columns are selected from a STDS map metadata view, if the user does not specify a subset (or at least one group column). `columns` is expected to be either `None` or a SQL compliant string with column names separated by comma (e.g. `id,start_time,end_time,semantic_label,creation_time`. So, if columns at this point is _None_, _*_ should be used in the query (available columns can vary between different types of STDS).